### ISSUE TITLE
asyncresolv won't build on Debian

### ISSUE THREAD

**Issue by lmoureaux** [lpetre-ulb/gemsuperbuild#3]
asyncresolv in xDAQ uses `-Werror` but triggers `unused-result` warnings that terminate the build. It could be solved by setting `CXXFLAGS=-Wno-error=unused-result` if asynreslov didn't use the following construct in its `configure.in`:

```cmake
# Warning options 
CXXFLAGS="${CXXFLAGS} -Werror -Wall -Winline"
```
I propose that we change it to:
```cmake
# Warning options 
CXXFLAGS="-Werror -Wall -Winline ${CXXFLAGS}"
```
This way `-W` flags we can be overriden from the environment.

**Comment by lpetre-ulb** [comment]
I agree with the bug, but I'm not convinced with the proposed solution as a first try. It will require patch to the sources and `CXXFLAGS` applied to the full `xDAQ` compilation. :-/

As it compiles without errors on CC7 and Archlinux, I would first try to investigate why the compilation fails on Debian. Maybe some hardening options such as `FORTIFY_SOURCE` are enabled by default. The error lies in a `read()` function call which is not usually marked as `warn_unused_result`.

**Comment by lmoureaux** [comment]
As you mentioned in private, this seems to be indeed caused by `FORTIFY_SOURCE`. There is no way around it other than disabling `-Werror=unused-result`.

A solution would be to replace `-Wall` by the list of `-Wxxx` flags that it enabled when the library was written. This would respect the intent of the original developer. Another would be to drop `-Werror` altogether. After seeing this problem, I don't think that `-Werror` should be used in release mode anyway...

Patching `configure.in` is in any case needed.

**Comment by lpetre-ulb** [comment]
Have you tried to set `CXXFLAGS=-U_FORTIFY_SOURCE` and/or `CXXFLAGS=-D_FORTIFY_SOURCE=0 ` ?

Though, that might not be the best solution since it disables the functionality for all `xDAQ` pacakges. I agree with you, `-Werror` should not be used in release mode. Can you provide a patch for `configure.in` since I don't have an easy access to a Debian machine? Or a PR ?

Anyway, I'm curious about the output of `g++ -Q -v <dummy>.cpp` on Debian. Could you provide me the output of the command ? Maybe in a private discussion.

**Comment by lpetre-ulb** [comment]
Fixed by #9.
